src/main/java/org/apache/ibatis/type/TypeHandlerRegistry.java

```diff
@@ -241,7 +241,8 @@ public boolean hasTypeHandler(TypeReference<?> javaTypeReference, JdbcType jdbcT
     }
     if (jdbcHandlerMap == null && type instanceof Class) {
       Class<?> clazz = (Class<?>) type;
-      if (clazz.isEnum()) {
+      if (clazz.isEnum() || (clazz.isAnonymousClass() && clazz.getEnclosingClass().isEnum())) {
+        clazz = clazz.isAnonymousClass() ? clazz.getEnclosingClass() : clazz;
```

Comment:
`getSuperclass()` seems to be a better choice here.

```suggestion
        clazz = clazz.isAnonymousClass() ? clazz.getSuperclass() : clazz;
```

Please let me know if I am missing something!

Reply:
...because `getEnclosingClass()` could throw SecurityException in Java 11.

Reply:
Fixed via c6672b5d0acdc89b28834a4bb17248ba7435e22a